src/Core/Worker/Cron.php

```diff
@@ -159,8 +159,13 @@ private static function cleanWorkerQueue()
 	 */
 	private static function deliverPosts()
 	{
-		$deliveries = DBA::p("SELECT `item-uri`.`uri` AS `inbox`, MAX(`failed`) AS `failed` FROM `post-delivery` INNER JOIN `item-uri` ON `item-uri`.`id` = `post-delivery`.`inbox-id` GROUP BY `inbox`");
+		$deliveries = DBA::p("SELECT `item-uri`.`uri` AS `inbox`, MAX(`failed`) AS `failed` FROM `post-delivery` INNER JOIN `item-uri` ON `item-uri`.`id` = `post-delivery`.`inbox-id` GROUP BY `inbox` ORDER BY RAND()");
```

Comment:
Why don't you order by `MAX(`failed`)`, this way the inboxes that didn't have delivery failures will be tried before the ones that failed at least once, etc...
```suggestion
		$deliveries = DBA::p("SELECT `item-uri`.`uri` AS `inbox`, MAX(`failed`) AS `failed` FROM `post-delivery` INNER JOIN `item-uri` ON `item-uri`.`id` = `post-delivery`.`inbox-id` GROUP BY `inbox` ORDER BY `failed`");
```

Reply:
We only directly deliver when there are no failures for that inbox. For the other cases we only make sure that there are worker jobs and that their priority matches the number of delivery attempts. The random makes sure that the deliver attempts are distributed evenly across the inboxes. For example you could have got a server starting with "a" who is really slow when receiving posts. This could result in a situation where this worker job would be killed since it exceed its lifetime. This would mean that inboxes that start with "z" would possibly never served.

Reply:
Ok.